daphne_worker/src/roles/aggregator.rs

```diff
@@ -410,40 +413,38 @@ impl<'srv> DapAggregator<DaphneWorkerAuth> for DaphneWorker<'srv> {
         Ok(!agg_share.empty())
     }
 
-    async fn put_out_shares(
+    async fn try_put_agg_share_span(
         &self,
         task_id: &TaskId,
         task_config: &DapTaskConfig,
-        part_batch_sel: &PartialBatchSelector,
-        out_shares: Vec<DapOutputShare>,
-    ) -> std::result::Result<HashSet<ReportId>, DapError> {
+        agg_share_span: DapAggregateShareSpan,
+    ) -> std::result::Result<Option<HashSet<ReportId>>, DapError> {
         let task_id_hex = task_id.to_hex();
         let durable = self.durable();
-        let mut agg_store_request_data: HashMap<String, Vec<DapOutputShare>> = HashMap::new();
+        let mut agg_store_request_data: HashMap<String, DapAggregateShare> = HashMap::new();
         let mut reports_processed_request_data: HashMap<String, Vec<ReportId>> = HashMap::new();
-        for (bucket, out_shares) in task_config
-            .as_ref()
-            .batch_span_for_out_shares(part_batch_sel, out_shares)?
-        {
-            for out_share in out_shares.into_iter() {
+        for (bucket, (aggregate_share, report_metadatas)) in agg_share_span {
+            for (id, time) in report_metadatas {
                 let reports_processed_name = self.config().durable_name_report_store(
                     task_config.as_ref(),
                     &task_id_hex,
-                    &out_share.report_id,
-                    out_share.time,
+                    &id,
+                    time,
                 );
                 reports_processed_request_data
                     .entry(reports_processed_name)
                     .or_default()
-                    .push(out_share.report_id.clone());
+                    .push(id.clone());
+            }
+            let agg_store_name =
+                durable_name_agg_store(&task_config.version, &task_id_hex, &bucket);
 
-                let agg_store_name =
-                    durable_name_agg_store(&task_config.version, &task_id_hex, &bucket);
+            debug_assert!(
                 agg_store_request_data
-                    .entry(agg_store_name)
-                    .or_default()
-                    .push(out_share);
-            }
+                    .insert(agg_store_name, aggregate_share)
+                    .is_none(),
+                "agg_store_names should be unique"
+            );
```

Comment:
We don't expect the agg_store_names to be unique. I think what we want to do is merge `agg_share` into the request data.